src/jdk.zipfs/share/classes/module-info.java

```diff
@@ -293,6 +293,22 @@
  *            .forEach(System.out::println);
  * }
  * </pre>
+ *
+ * @implNote
+ * While the
+ * <a href="https://support.pkware.com/pkzip/appnote">ZIP specification</a>
+ * has no restrictions on ZIP entries with duplicate names, the
+ * {@link ZipOutputStream} class doesn't allow creation of ZIP archives with
+ * duplicate entry names. Therefore, when parsing archives containing duplicate
+ * names, unexpected behavior may occur such as metadata or content from the
+ * later entry being returned due to internal caching.
+ *
+ * Additionally, for compatibility with earlier versions of the JDK, files and
+ * directories with the same name (such as "foobar" and "foobar/") are
+ * considered duplicates, which can result in unexpected behavior such as wrong
+ * metadata or content being returned when parsing archives containing
+ * duplicate names.
```

Comment:
The jdk.zipfs module is the JDK's ZIP file system provider, it allows the contents of a ZIP or JAR file be treated as a file system. I think what is missing from this module description is the mapping of file system concepts to ZIP files. If we added a section to the documentation for that then it would be a place to document the file separator, normalization, how directories and file tree are synthesized, and the limitation that a file system cannot be constructed when it leads to have several entries in a directory having equal names. There's a lot more that could be added but hopefully you can see that an implNote and warnings just doesn't work as a way to present the mapping.

Reply:
> That would probably also involve taking existing documentation such as the note about not opening entries with "."/"..", and the POSIX permissions mappings? Would it make sense to split the rest of the changes in this PR from the zipfs changes since that's going to be a bigger undertaking?

Yes, I think it should be separated, and it would be nice to drop the changes to jdk.zipfs from this PR. If/when this documentation is written then it describe the "/" character and the synthesized links that are "." and "..", which can lead to documentation that a zip file that contain contains with  "." or ".." elements cannot be opened as a file system.